tooling/release_download_test.sh

```diff
@@ -8,66 +8,191 @@
+
+"
+  echo "$USAGE"
+  exit 1
+}
+
+while getopts ":hvks" opt; do
+    case "${opt}" in
+        h)
+            usage
+            ;;
+        v)
+            VERBOSE=true
+            ;;
+        k)
+            KEEP_STAGING=true
+            ;;
+        s)
+            SKIP_DOWNLOADING=true
+            ;;
+        "?")
+            echo "Unknown option '-$OPTARG'"
+            usage
+            ;;
+        ":")
+            echo "No argument value for option '-$OPTARG'"
+            usage
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))
 
 [ "$VERBOSE" = "true" ] && set +x
-if echo "$TAG" | grep jdk8u; then
+
+if [ $# -ne "1" ]; then
+    usage
+fi
+
+# the tag should be the remaining argument
+TAG=${1}
+
+if [ -z "${TAG-}" ]; then
+    usage
+fi
+
+print_verbose "IVT : Verifying Tag '$TAG'"
+
+##########################################################################################################################
+#
+# Extract JDK major version from TAG.
+#
+##########################################################################################################################
+
+if echo "$TAG" | grep jdk8u > /dev/null; then
   MAJOR_VERSION=8
-elif echo "$TAG" | grep ^jdk-; then
+elif echo "$TAG" | grep ^jdk- > /dev/null; then
   MAJOR_VERSION=$(echo "$TAG" | cut -d- -f2 | cut -d. -f1 | cut -d\+ -f1)
 else
   # Probably a beta with the tag starting jdkXXu
   MAJOR_VERSION=$(echo "$TAG" | cut -d- -f1 | tr -d jdku)
 fi
 
-echo "$(date +%T) : IVT : I will be checking https://github.com/adoptium/temurin${MAJOR_VERSION}-binaries/releases/tag/$TAG"
+print_verbose "IVT : I will be checking https://github.com/adoptium/temurin${MAJOR_VERSION}-binaries/releases/tag/$TAG"
 if [ -z "${MAJOR_VERSION}" ] || [ -z "${TAG}" ]; then
-   echo "MAJOR_VERSION or TAG undefined - aborting"
+   print_error "MAJOR_VERSION or TAG undefined - aborting"
    exit 1
 fi
 
+##########################################################################################################################
+#
+# Download release information from GitHub for the specified major version.
+#
+##########################################################################################################################
+
 if ! curl -sS "https://api.github.com/repos/adoptium/temurin${MAJOR_VERSION}-binaries/releases" > "$WORKSPACE/jdk${MAJOR_VERSION}.txt"; then
-   echo "github API call failed - aborting"
+   print_error "GitHub API call failed - aborting"
    exit 2
 fi
 
-[ "$VERBOSE" = "true" ] && echo "$(date +%T) : IVT: Downloading files from release repository"
+##########################################################################################################################
+#
+# Download all files of the specified release if required.
+#
+##########################################################################################################################
+
+if [ "$SKIP_DOWNLOADING" = "false" ]; then
```

Comment:
could this be a function?

Reply:
same goes for several commented blocks below.

Reply:
indeed, I was thinking the same that the various blocks should be extracted into functions, will do so.

Reply:
I extracted most of the code into separate functions.